submariner/ipsec-certificate-support.md

```diff
@@ -0,0 +1,86 @@
+# Submariner Enhancement Proposal: Certificate Mode for OVN-Kubernetes IPsec
+
+## Summary
+
+This enhancement enables Submariner to **share the Pluto (Libreswan) process** with OpenShift/OVN-Kubernetes when OVN
+IPsec is enabled. The primary goal is to allow both Submariner and OVN-Kubernetes to provide secure, encrypted inter-cluster
+and intra-cluster networking on the same OpenShift cluster, without process conflicts or network disruption.
+
+To achieve this, Submariner reuses the existing Pluto process started by OVN-Kubernetes, rather than starting its own.
+Because OVN-Kubernetes configures Pluto to use X.509 certificates for authentication, Submariner must also operate in
+**certificate mode** as a necessary side effect.
+
+To achieve this, Submariner is enhanced to:
+
+- Reuse the existing Pluto (Libreswan) process started by OVN-Kubernetes, rather than starting its own.
+- Operate in certificate mode, matching OVN’s use of X.509 certificates for authentication.
+- Provide an explicit deployment option to enable this mode, ensuring that Submariner and OVN-Kubernetes can both manage
+IPsec tunnels independently but harmoniously.
+
+---
+
+## Motivation
+
+- **Problem:**
+  When both Submariner and OVN-Kubernetes IPsec are enabled, each tries to start its own Pluto process, causing conflicts
+and breaking cluster networking.
+- **Goal:**
+  Allow Submariner and OVN-Kubernetes to run together, each managing their own tunnels, but sharing
+the same Pluto process and certificate-based authentication infrastructure.
+
+---
+
+## Design Details
+
+### Manual Enablement
+
+- Certificate mode must be **explicitly enabled** by passing an option (e.g., `--ipsec-cert-mode`) to `subctl` during
+deployment or join.
+
+### Pluto Process Expectation
+
+- When certificate mode is enabled, Submariner **expects** that a compatible Pluto process (as started by OVN-Kubernetes)
+is already running at the required path (`/usr/local/bin/pluto` by default).
+- The Pluto control socket is expected at `/run/pluto/pluto.ctl`.
```

Comment:
Why is the daemon’s path important? If Submariner can find the control socket, that should be enough, shouldn’t it?